interface/patient_file/encounter/find_code_dynamic_ajax.php

```diff
@@ -0,0 +1,333 @@
+    $feitem['fld_rows'   ] = 0;
+    $key = $column ? 'title' : 'field_id';
+    $arr[$feitem[$key]] = $feitem;
+  }
+  ksort($arr);
+  if ($reverse) $arr = array_reverse($arr);
+  return $arr;
+}
+
+function genFieldIdString($row) {
+  return 'CID|' . json_encode($row);
+}
+
+// Column sorting parameters.
+//
+$orderby = '';
+$fe_column = 0;
+$fe_reverse = false;
+if (isset($_GET['iSortCol_0'])) {
+	for ($i = 0; $i < intval($_GET['iSortingCols']); ++$i) {
+    $iSortCol = intval($_GET["iSortCol_$i"]);
+		if ($_GET["bSortable_$iSortCol"] == "true" ) {
+      $sSortDir = escape_sort_order($_GET["sSortDir_$i"]); // ASC or DESC
+      // We are to sort on column # $iSortCol in direction $sSortDir.
+      $orderby .= $orderby ? ', ' : 'ORDER BY ';
+
+      // Note the primary sort column and direction for later logic.
+      if ($i == 0) {
+        $fe_column = $iSortCol;
+        $fe_reverse = $sSortDir == 'DESC';
+      }
+
+      if ($what == 'codes') {
+        if ($iSortCol == 0) {
+          $orderby .= $prod ? "d.drug_id $sSortDir, t.selector $sSortDir" : "c.code $sSortDir";
+        }
+        else {
+          $orderby .= $prod ? "d.name $sSortDir" : "c.code_text $sSortDir";
+        }
+      }
+      else if ($what == 'fields') {
+        if ($source == 'V') {
+          // No action needed here.
+        }
+        else {
+          // Remaining sources (D, H, E) come from a layout.
+          if ($iSortCol == 0) {
+            $orderby .= "lo.field_id $sSortDir";
+          }
+          else {
+            $orderby .= "lo.title $sSortDir";
+          }
+        }
+      }
+      else if ($what == 'lists') {
+        if ($iSortCol == 0) {
+          $orderby .= "li.list_id $sSortDir";
+        }
+        else {
+          $orderby .= "li.option_id $sSortDir";
+        }
+      }
+      else if ($what == 'groups') {
+        if ($iSortCol == 0) {
+          $orderby .= "code $sSortDir";
+        }
+        else {
+          $orderby .= "description $sSortDir";
+        }
+      }
+		}
+	}
+}
+
+if ($what == 'codes') {
+  $sellist = $prod ?
+    "CONCAT(d.drug_id, '|', COALESCE(t.selector, '')) AS code, d.name AS description, '$codetype' AS codetype" :
+    "CONCAT(c.code, '|') AS code, c.code_text AS description, '$codetype' AS codetype";
+  $where1 = '';
+  $where2 = '';
+  if ($prod) {
+    $from = "drugs AS d LEFT JOIN drug_templates AS t ON t.drug_id = d.drug_id";
+    if (!$include_inactive) $where1 = "WHERE d.active = 1";
+  }
+  else {
+    $from = "codes AS c";
+    $where1 = "WHERE c.code_type = '$ncodetype'";
+    if (!$include_inactive) $where1 .= " AND c.active = 1";
+  }
+  if (isset($_GET['sSearch']) && $_GET['sSearch'] !== "") {
+    $sSearch = add_escape_custom($_GET['sSearch']);
+    $where2 = empty($where1) ? "WHERE " : " AND ";
+    $where2 .= ($prod ?
+      "(d.name LIKE '%$sSearch%' OR t.selector LIKE '%$sSearch%')" :
+      "(c.code LIKE '%$sSearch%' OR c.code_text LIKE '%$sSearch%')");
+  }
+}
```

Comment:
Are these for codes that aren't external type (suchs as ICD9/ICD10/SNOMED etc.)

Reply:
Also need to implement sql binding or sql escaping in above and below code blocks

Reply:
Looks like everything is already escaped except $layout_id which I will fix.